src/coins.h

```diff
@@ -147,20 +165,32 @@ struct CCoinsCacheEntry
     inline void AddFlags(uint8_t flags, CoinsCachePair& self, CoinsCachePair& sentinel) noexcept
     {
         Assume(&self.second == this);
+        if (!m_flags && flags) {
+            m_prev = sentinel.second.m_prev;
+            m_next = &sentinel;
+            sentinel.second.m_prev = &self;
+            m_prev->second.m_next = &self;
+        }
         m_flags |= flags;
     }
     inline void ClearFlags() noexcept
     {
+        if (!m_flags) return;
+        m_next->second.m_prev = m_prev;
```

Comment:
An earlier version of this PR that relied on checking for null did do this. I just think it's unnecessary work to set them to null, because access is guarded by `m_flags`.